### PR TITLE
feat(generate:typetests): Move typetest config to external file

### PR DESCRIPTION
**Pull request by tylerbutler**
Typetest config is stored in package.json today, but this causes a lot of churn in package.json when we release and typetests are updated. This PR tries to load the type test config from a type-validation.config.json file so it's separate from package.json. Config will fall back to loading from package.json if that file is missing.

This should help main/next integrations be a little simpler. When merging from main to next, for the typetest files either side can be accepted, because re-running generation with the `--branch` argument will always write the correct values for next.

BREAKING CHANGE: We now prefer external files for the typetest config, so when running, it will always output to the external config file, and will remove the package.json typeValidation node.

### CODE REVIEW

File: build-tools/packages/build-tools/src/typeValidator/packageJson.ts [hunk 1]
```diff
@@ -49,13 +54,24 @@ function safeParse(json: string, error: string) {
...
+		packageProp: "typeValidation",
+		stopDir: resolve(packageDir),
+	});
 
+	// use load instead of search because we don't want to recurse up the directory tree looking for configs
+	const result =
+		configExplorer.load("typeValidation.config.json") ?? configExplorer.load("package.json");
+	const config = result?.config;
+	console.log(`loaded from: ${result?.filepath}`);
+
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/typeValidator/packageJson.ts:73]:
```suggestion

```

File: build-tools/packages/build-tools/src/typeValidator/packageJson.ts [hunk 2]
```diff
@@ -464,25 +479,34 @@ export async function getAndUpdatePackageDetails(
...
-			packageDetails.json.typeValidation.disabled = disabled;
+			packageDetails.typeValidation.disabled = disabled;
 		}
 
 		if ((writeUpdates ?? false) === true) {
-			await util.promisify(fs.writeFile)(
-				`${packageDir}/package.json`,
-				JSON.stringify(packageDetails.json, undefined, 2).concat("\n"),
+			await writeJson(
+				`${packageDir}/typeValidation.config.json`,
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/typeValidator/packageJson.ts:498]: Nit: Seems like most json config files are named using kebab case (package-lock, api-extractor, etc.). Should we considfer `type-validation.config.json`?

File: build-tools/packages/build-tools/src/typeValidator/packageJson.ts [hunk 3]
```diff
@@ -49,13 +54,22 @@ function safeParse(json: string, error: string) {
...
+	if (!existsSync(packagePath)) {
 		throw new Error(`Package json does not exist: ${packagePath}`);
 	}
-	const content = await util.promisify(fs.readFile)(packagePath);
-
-	const pkgJson: PackageJson = safeParse(content.toString(), packagePath);
 
+	// try loading from typeValidation.config.json, then fall back to package.json
+	const configExplorer = cosmiconfigSync("typeValidation", {
+		searchPlaces: ["typeValidation.config.json", "package.json"],
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/typeValidator/packageJson.ts:63]: Nit: these file name strings are used in a few places. Should we create variables for them?

File: build-tools/packages/build-tools/src/common/fluidRepo.ts [hunk 4]
```diff
@@ -54,6 +54,8 @@ export interface IFluidBuildConfig {
 	branchReleaseTypes?: {
 		[name: string]: VersionBumpType | PreviousVersionStyle;
 	};
+
+	typeTests?: ITypeValidationConfig;
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/common/fluidRepo.ts:58]: Docs? 🙂

**Review by Josmithr — APPROVED** [review]
I love these changes. I think they're make the main -> next process a lot easier, and isolate diffs better as well.

File: build-tools/packages/build-tools/src/typeValidator/packageJson.ts (hunk 3 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/typeValidator/packageJson.ts:63]: Done.

File: build-tools/packages/build-tools/src/typeValidator/packageJson.ts (hunk 2 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/typeValidator/packageJson.ts:498]: Done in latest.

**Pull request merged**